lib/ArangoDBClient/Document.php

```diff
@@ -19,7 +19,7 @@
  * @package   ArangoDBClient
  * @since     0.2
  */
-class Document
```

Comment:
How did that get in? :smiley:

Implementation of JsonSerializable was rejected in PR #206 due to it adding a 70-100% overhead to object instantiation and handling (which does make a difference in a tight loop) and a way already exists to get the json-encoded data back (see https://github.com/arangodb/arangodb-php/pull/206#issuecomment-279149315). 

However, since your proposed changes allow subclassing, one has the choice to implement the JsonSerializer in a ny subclasses of `Document`, used. That gives people the choice to keep a more lean base class, avoiding said overhead.